mindsdb/integrations/handlers/anyscale_endpoints_handler/anyscale_endpoints_handler.py

```diff
@@ -0,0 +1,175 @@
+import os
+import json
+import openai
+import contextlib
+from typing import Optional, Dict
+
+import pandas as pd
+
+from mindsdb.integrations.handlers.openai_handler.openai_handler import OpenAIHandler
+from mindsdb.integrations.handlers.openai_handler.constants import OPENAI_API_BASE
+
+
+# TODO: retrieve these from API once possible
+CHAT_MODELS = (
+    'meta-llama/Llama-2-7b-chat-hf',
+    'meta-llama/Llama-2-13b-chat-hf',
+    'meta-llama/Llama-2-70b-chat-hf',
+    'codellama/CodeLlama-34b-Instruct-hf',
+)
+ANYSCALE_API_BASE = 'https://api.endpoints.anyscale.com/v1'
+
+
+class AnyscaleEndpointsHandler(OpenAIHandler):
+    name = 'anyscale_endpoints'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.all_models = list(CHAT_MODELS)
+        self.chat_completion_models = CHAT_MODELS
+        self.supported_ft_models = CHAT_MODELS  # base models compatible with fine-tuning
+        self.default_model = CHAT_MODELS[0]
+        self.base_api = ANYSCALE_API_BASE
+        self.default_mode = 'default'  # can also be 'conversational' or 'conversational-full'
+        self.supported_modes = ['default', 'conversational', 'conversational-full']
+        self.rate_limit = 25  # requests per minute
+        self.max_batch_size = 20
+        self.default_max_tokens = 100
+        self.ft_cls = openai.FineTuningJob  # non-legacy finetuning endpoint
+
+    @staticmethod
+    @contextlib.contextmanager
+    def _anyscale_base_api(key='OPENAI_API_BASE'):
+        """ Temporarily updates the API base env var to point towards the Anyscale URL. """
+        old_base = os.environ.get(key, OPENAI_API_BASE)
+        os.environ[key] = ANYSCALE_API_BASE
+        try:
+            yield  # enter
+        finally:
+            os.environ[key] = old_base  # exit
+
+    def create(self, target, args=None, **kwargs):
+        with self._anyscale_base_api():
+            super().create(target, args, **kwargs)
+
+    def predict(self, df: pd.DataFrame, args: Optional[Dict] = None) -> pd.DataFrame:
+        with self._anyscale_base_api():
+            return super().predict(df, args)
+
+    def finetune(self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        with self._anyscale_base_api():
+            super().finetune(df, args)
+            # rewrite chat_completion_models to include the newly fine-tuned model
+            args = self.model_storage.json_get('args')
+            self.chat_completion_models = list(self.chat_completion_models) + [args['model_name']]
```

Comment:
@ea-rus just for confirmation (because this does work empirically for `n=1` users): the stateful class instances that our executor runs is at each user's level, correct? I.e. modifying this list with model `X` won't leak to other users who may have finetuned model `Y`?

Reply:
I don't know context much. Can the same model name be used by other user? If yes, it should have some unique part in name, for example model_name = <predictor id> + model name
 
Also question: is chat_completion_models variable persisted anywhere? if not it will be cleared after handler reinitialized

Reply:
> Can the same model name be used by other user?
Because it is a unique model fine-tuned by this user, we want to avoid that situation. There is a unique part in the name, the handler leverages logic for this in the OpenAI handler.

> It will be cleared after handler reinitialized
Nice, this is what my original question aimed at. Thanks for clarifying. I will have to modify this slightly to store all finetuned model names per user.